pytorch_lightning/utilities/parsing.py

```diff
@@ -177,47 +177,46 @@ def __repr__(self):
 def lightning_hasattr(model, attribute):
     """ Special hasattr for lightning. Checks for attribute in model namespace,
         the old hparams namespace/dict, and the datamodule. """
-    trainer = model.trainer
-
-    # Check if attribute in model
-    if hasattr(model, attribute):
-        attr = True
-    # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
-        if isinstance(model.hparams, dict):
-            attr = attribute in model.hparams
-        else:
-            attr = hasattr(model.hparams, attribute)
-    # Check if the attribute in datamodule (datamodule gets registered in Trainer)
-    elif trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
-        attr = getattr(trainer.datamodule, attribute)
-    else:
-        attr = False
+    try:
+        lightning_getattr(model, attribute)
+    except AttributeError:
+        return False
 
-    return attr
+    return True
 
 
 def lightning_getattr(model, attribute):
     """ Special getattr for lightning. Checks for attribute in model namespace,
         the old hparams namespace/dict, and the datamodule. """
-    trainer = model.trainer
+
+    trainer = getattr(model, 'trainer', None)
+    datamodule = getattr(trainer, 'datamodule', None)
 
     # Check if attribute in model
     if hasattr(model, attribute):
         attr = getattr(model, attribute)
+
     # Check if attribute in model.hparams, either namespace or dict
     elif hasattr(model, 'hparams'):
         if isinstance(model.hparams, dict):
-            attr = model.hparams[attribute]
+            try:
+                attr = model.hparams[attribute]
+            except KeyError:
+                raise AttributeError(
```

Comment:
Seems too early to raise? (datamodule attr checking is below this)

Reply:
Thank you for reviewing! As pointed out, it's too early to raise an error. I'll fix the problem.

Reply:
To fix the problem, I used multiple return statements in 5f1cd56. I am neutral about using multiple return statements, so please let me know if I should avoid using them. [Stack Overflow: Should a function have only one return statement?](https://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement)